hedera-node/src/main/java/com/hedera/services/tokens/HederaTokenStore.java

```diff
@@ -446,6 +437,9 @@ public ResponseCodeEnum update(TokenManagement changes) {
 				token.setSymbol(newSymbol);
 				symbolKeyedIds.put(newSymbol, tId);
 			}
+			if (hasNewTokenName) {
+				token.setTokenName(changes.getTokenName());
```

Comment:
You must update the `tokenNameKeyedIds` mapping as-well otherwise the map will be outdated and not consistent. Similarly to the symbol update

Reply:
You need to update the `nameKeyedIds`.